app/controllers/application_controller.rb

```diff
@@ -6,7 +6,7 @@ class ApplicationController < ActionController::API
 
   def current_user
     @current_user ||= User.find(@device.resource_owner_id)          if request.headers['X-Physical-Secret']
-    @current_user ||= User.find(doorkeeper_token.resource_owner_id) if request.headers['Authorization']
+    @current_user ||= User.find(doorkeeper_token.resource_owner_id) if request.headers['Authorization'] or (not request.headers['X-Physical-Secret'] and doorkeeper_token)
```

Comment:
I would try to clean it up a bit more. 
This way the `'X-Physical-Secret` token whenever it is set, will have priority

```
...
@current_user ||=  User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token or request.headers['Authorization']
@current_user ||= User.find(@device.resource_owner_id) if request.headers['X-Physical-Secret']
```